src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -310,7 +310,7 @@ def _validate_vm_create_storage_profile(namespace, for_scale_set=False):
                 forbidden.remove(prop)
 
     # set default storage SKU if not provided and using an image based OS
-    if not namespace.storage_sku and namespace.storage_profile not in [StorageProfile.ManagedSpecializedOSDisk, StorageProfile.SASpecializedOSDisk]:  # pylint: disable=line-too-long
+    if not namespace.storage_sku and namespace.storage_profile in [StorageProfile.SAPirImage, StorageProfile.SACustomImage]:  # pylint: disable=line-too-long
         namespace.storage_sku = 'Standard_LRS' if for_scale_set else 'Premium_LRS'
```

Comment:
For VMSS, the current design is the sku should always default to `standard_lrs`, as low-end configuration is sufficient for scaling out by throwing in more instances.  Please let me know otherwise.